openpype/hosts/houdini/plugins/create/create_vbd_cache.py

```diff
@@ -42,3 +53,53 @@ def get_network_categories(self):
             hou.ropNodeTypeCategory(),
             hou.sopNodeTypeCategory()
         ]
+
+    def get_sop_node_path(self, selected_node):
+        """Get Sop Path of the selected node.
+
+        Although Houdini allows ObjNode path on `sop_path` for the
+        the ROP node, we prefer it set to the SopNode path explicitly.
+        """
+
+        # Allow sop level paths (e.g. /obj/geo1/box1)
+        if isinstance(selected_node, hou.SopNode):
+            self.log.debug(
+                "Valid SopNode selection, 'SOP Path' in ROP will"
+                " be set to '%s'.", selected_node.path()
+            )
+            return selected_node.path()
+
+        # Allow object level paths to Geometry nodes (e.g. /obj/geo1)
+        # but do not allow other object level nodes types like cameras, etc.
+        elif isinstance(selected_node, hou.ObjNode) and \
+                selected_node.type().name() in ["geo"]:
+
+            # get the output node with the minimum
+            # 'outputidx' or the node with display flag
+            sop_node = self.get_obj_output(selected_node)
+            if sop_node:
+                self.log.debug(
+                    "Valid ObjNode selection, 'SOP Path' in ROP will "
+                    "be set to the child path '%s'.", sop_node.path()
+                )
+                return sop_node.path()
+
+    def get_obj_output(self, obj_node):
+        """Find output node with the smallest 'outputidx'."""
```

Comment:
The docstring isn't entirely true - because it doesn't necessarily find an output node.

If there is no output node, then it takes the node with a `Display flag` set as per `SopNode.subnetOutputs()` documentation:

> Certain networks, such as SOPs, have special Output nodes to override the canonical Display or Render nodes and allow multiple outputs. If any output nodes are present, this returns a list of those nodes. If no output nodes are present, either the display or render node is returned depending on if the target of cooking is the display or an output driver. If the node has no children, an empty list is returned.

I think it's a good thing that node is considered - but just wanted to remark that the docstring isn't entirely accurate.